src/python_testing/tc_mode_base.py

```diff
@@ -0,0 +1,112 @@
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from mobly import asserts
+from chip.clusters.Types import NullValue
+
+logger = logging.getLogger(__name__)
+
+
+class TC_MODE_BASE:
+
+    def initialize_tc_base(self, endpoint, attributes, requested_cluster, cluster_objects, *args):
+        self.commonTags = {0x0: 'Auto',
+                           0x1: 'Quick',
+                           0x2: 'Quiet',
+                           0x3: 'LowNoise',
+                           0x4: 'LowEnergy',
+                           0x5: 'Vacation',
+                           0x6: 'Min',
+                           0x7: 'Max',
+                           0x8: 'Night',
+                           0x9: 'Day'}
+        self.specificTags = [tag.value for tag in requested_cluster.Enums.ModeTag]
+        self.supported_modes_dut = set()
+        self.cluster_objects = cluster_objects
+        self.supported_modes = None
+        self.endpoint = endpoint
+        self.attributes = attributes
+
+    async def check_supported_modes_and_labels(self):
+        # Get the supported modes
+        self.supported_modes = await self.read_single_attribute_check_success(endpoint=self.endpoint,
+                                                                              cluster=self.cluster_objects,
+                                                                              attribute=self.attributes.SupportedModes)
+
+        # Check if the list of supported modes is larger than 2
+        asserts.assert_greater_equal(len(self.supported_modes), 2, "SupportedModes must have at least 2 entries!")
+        # Check that supported modes are less than 255
+        asserts.assert_less_equal(len(self.supported_modes), 255, "SupportedModes must have at most 255 entries!")
+
+        # Check for repeated labels or modes
+        labels = set()
+        for mode_options_struct in self.supported_modes:
+            # Verify that the modes in all ModeOptionsStruct in SupportedModes are unique.
+            if mode_options_struct.mode in self.supported_modes_dut:
+                asserts.fail("SupportedModes can't have repeated Mode values")
+            else:
+                self.supported_modes_dut.add(mode_options_struct.mode)
+            # Verify that the labels in all ModeOptionsStruct in SupportedModes are unique.
+            if mode_options_struct.label in labels:
+                asserts.fail("SupportedModes can't have repeated Label values")
+            else:
+                labels.add(mode_options_struct.label)
+
+    async def check_if_labels_in_lists(self, requiredtags=None):
```

Comment:
does this need to be async? looked briefly for some await in impl and did not see anything obvious.

Reply:
My mistake, this is actually not needed.